src/main/java/vswe/stevesfactory/components/ComponentMenuContainer.java

```diff
@@ -86,334 +86,334 @@ public ComponentMenuContainer(FlowComponent parent, ConnectionBlockType validTyp
-                    return prefix;
-                }
-
-                @SideOnly(Side.CLIENT)
-                @Override
-                public List<String> getSuffix(GuiBase gui)
-                {
-                    return locked ? lockedSuffix : suffix;
-                }
-            }
-
-
-            @SideOnly(Side.CLIENT)
-            @Override
-            public void drawMouseOver(GuiManager gui, int mX, int mY)
-            {
-                if (locked && GuiBase.isShiftKeyDown())
-                {
-                    drawMouseOver(gui, cachedContainer, lockedX, lockedY, mX, mY);
-                    cachedTooltip.drawMouseOverMouseOver(gui, lockedX + gui.getAdvancedToolTipContentStartX(cachedTooltip), lockedY + gui.getAdvancedToolTipContentStartY(cachedTooltip), mX, mY);
-                } else
-                {
-                    locked = false;
-                    keepCache = false;
-                    super.drawMouseOver(gui, mX, mY);
-                    if (!keepCache)
-                    {
-                        cachedTooltip = null;
-                        cachedContainer = null;
-                    }
-                }
-            }
-
-            @SideOnly(Side.CLIENT)
-            @Override
-            protected void drawMouseOver(GuiManager gui, IContainerSelection iContainerSelection, int mX, int mY)
-            {
-                drawMouseOver(gui, iContainerSelection, mX, mY, mX, mY);
-            }
-
-            @SideOnly(Side.CLIENT)
-            private void drawMouseOver(GuiManager gui, IContainerSelection iContainerSelection, int x, int y, int mX, int mY)
-            {
-                boolean isBlock = !iContainerSelection.isVariable();
-
-                if (GuiScreen.isShiftKeyDown() && isBlock)
-                {
-                    if (cachedTooltip == null || cachedId != iContainerSelection.getId())
-                    {
-                        cachedContainer = iContainerSelection;
-                        cachedTooltip = new ToolTip(gui, (ConnectionBlock) iContainerSelection);
-                        cachedId = iContainerSelection.getId();
-                    }
-                    keepCache = true;
-
-                    gui.drawMouseOver(cachedTooltip, x, y, mX, mY);
-                } else
-                {
-                    List<String> lines = getMouseOverForContainer(iContainerSelection, selectedInventories);
-                    if (isBlock)
-                    {
-                        if (lines == null)
-                        {
-                            lines = new ArrayList<String>();
-                        }
-
-                        lines.add("");
-                        lines.add(Color.GRAY + Localization.TOOLTIP_EXTRA_INFO.toString());
-                    }
-
-                    gui.drawMouseOver(lines, mX, mY);
-                }
-            }
-        };
+//        scrollController = new ScrollController<IContainerSelection>(getDefaultSearch())
```

Comment:
This is fucking horrible, this could be the main cause of the issue. Bad...VERY bad.